routing/router.go

```diff
@@ -173,7 +172,13 @@ type Config struct {
 	// payment was unsuccessful.
 	SendToSwitch func(firstHop lnwire.ShortChannelID, paymentID uint64,
 		htlcAdd *lnwire.UpdateAddHTLC,
-		circuit *sphinx.Circuit) ([sha256.Size]byte, error)
+		circuit *sphinx.Circuit) error
+
+	// GetPaymentResult returns the the result of the payment attempt with
+	// the given paymentID. This method blocks until the result is
+	// available, or an error is encountered. If the paymentID is unknown,
+	// htlcswitch.ErrPaymentIDNotFound will be returned.
+	GetPaymentResult func(paymentID uint64) (htlcswitch.PaymentResult, error)
```

Comment:
Perhaps we should combine this closure and the above method into a new interface? Possible names are: `router.FirstHopDispatcher`? kek

Reply:
hm, for easy mocking?